explorer/testdata/template/fail_no_member.carbon

```diff
@@ -0,0 +1,22 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// RUN: %{not} %{explorer-run}
+// RUN: %{not} %{explorer-run-trace}
+
+package ExplorerTest api;
+
+interface GetX {
+  fn DoIt[self: Self]() -> i32;
+}
+
+impl forall [template T:! type] T as GetX {
+  // CHECK:STDERR: COMPILATION ERROR: {{.*}}/explorer/testdata/template/fail_no_member.carbon:[[@LINE+1]]: member access, unexpected i32 in self.x
```

Comment:
This error is kind of confusing; where in `self.x` is the unexpected `i32`? This looks like you're falling through to the "unhandled" case of TypeCheckExp, not a deliberately written error message.

Reply:
i.e., handling this should probably be part of this, since it seems you may be adding code that deliberately reaches an unhandled code path.

Reply:
This isn't new in this PR: https://carbon.godbolt.org/z/f87nc4e4f

(It would be nice to include a note saying that we're performing an instantiation with T=i32; there's a TODO in the code for that already.)

Reply:
Do you think this would make sense for a "good first issue"?

Reply:
Sure, filed as #2705.